src/random_test.cpp

```diff
@@ -2716,6 +2721,10 @@ bool execute_sql(const std::string &sql, Thd1 *thd) {
   if (res != 0) { // query failed
     thd->failed_queries_total++;
     thd->max_con_fail_count++;
+    // Manage recent queries for failed queries
+    if (!log_N) {
```

Comment:
the condition should be if(options->at(Option::LOG_N_QUERIES)->getInt()>0)

Reply:
added